packages/babel-traverse/src/path/conversion.ts

```diff
@@ -195,24 +195,44 @@ export function arrowFunctionToExpression(
  * or "new.target", ensure that these references reference the parent environment around this function.
  */
 function hoistFunctionEnvironment(
-  fnPath,
+  fnPath: NodePath<t.Function>,
   // TODO(Babel 8): Consider defaulting to `false` for spec compliancy
   noNewArrows = true,
   allowInsertArrow = true,
-) {
-  const thisEnvFn = fnPath.findParent(p => {
+): { thisBinding: string; fnPath: NodePath<t.Function> } {
```

Comment:
Can you add some comments about the return type? Especially we may have different `fnPath` output than the input.